src/routes/safe/components/Balances/SendModal/screens/ContractInteraction/ContractABI/index.tsx

```diff
@@ -19,12 +23,40 @@ const mustBeValidABI = (abi: string): undefined | string => {
   }
 }
 
-const ContractABI = () => (
-  <Row margin="sm">
-    <Col>
-      <TextareaField name="abi" placeholder="ABI*" text="ABI*" type="text" validate={mustBeValidABI} />
-    </Col>
-  </Row>
-)
+const ContractABI = (): React.ReactElement => {
+  const {
+    input: { value: contractAddress },
+  } = useField('contractAddress', { subscription: { value: true } })
+  const { mutators } = useForm()
+  const setAbiValue = React.useRef(mutators.setAbiValue)
+
+  React.useEffect(() => {
+    const validateAndSetAbi = async () => {
+      if (!mustBeEthereumAddress(contractAddress) && !(await mustBeEthereumContractAddress(contractAddress))) {
```

Comment:
I'm the only one having real troubles with this naming convention? 

Each time I find this `mustBe...` with the negation I have to think twice that double negation in semantic is necessary to understand this.

Reply:
Yes, you're not alone.

But, for the record, they are inherited from `final-form` validator definitions.

You can see in the documentation itself https://final-form.org/docs/react-final-form/examples/field-level-validation

If the function returns `undefined`, it's a valid value, if not it should return a string with the error message.

Maybe we should wrap those in another function that returns a boolean if we're going to use those validators outside the form. WDYT?

Reply:
yes Fer, I'm with you on that wrapper. Should be handled this in this PR? or create a new one and replace all the occurrences for these cases?